src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden && animal.health > 0) {
+      animal.hide();
+      animal.health -= 50;
+
+      if (animal.health <=0) this.die();
```

Comment:
[CODE PERFORMANCE] - Instead of calling a non-existing `die` method, you should filter the `Animal.alive` array to remove the dead animal. Use the `filter` method to remove any animals whose health is less than or equal to `0`. This will avoid the need for a `die` method and adhere to the checklist requirement to use filter instead of nested loops.